src/array.cr

```diff
@@ -1770,4 +1770,15 @@ class Array(T)
       end
     end
   end
+
+  # This method returns the i-th element. If i is equal to ary.size, it returns nil.
+  def bsearch(&block : T -> Int|Float|Bool?)
+    idx=bsearch_index(&block)
+    idx ? self[idx] : nil
```

Comment:
I would vote for two overloads

* `bsearch(&block)` which raises if not found
* `bsearch?(default=nil, &block)` which return the default if not found.

That way we keep away the Nils